### ISSUE TITLE
Invalid Bundle Structure caused by .so files when uploading to App Store

### ISSUE THREAD

**Issue by Dirk-Sandberg** [kivy/kivy-ios#306]
Hi Guys,

I've ben struggling with this problem for a while now. My app works great when running on the iPhone simulator and my personal phone, but I can't upload it to the App Store. It's because of the '.so' files in the default 'YourApp' directory that is made by kivy-ios. This 'YourApp' directory is in the Copy Bundle Resources section of the "Build Phases" tab in Xcode, which apparently can't hold .so files when uploading to the App Store.

Similar questions online say to move the problematic files from the Copy Bundle Resources section to the Link Binary With Libraries section, but the program fails to run even on the iPhone simulator if I do that in this case.

 The '.so' files its complaining about are from the modules I use in my program like kivy, requests, PIL etc. I have those modules (the folder containing all their code) inside my project's folder because that's the only where I could get kivy-ios to find the modules I needed so it can package it -- perhaps there's a better place to put required modules? Here's the error I get when trying to upload to the App Store:

<img width="752" alt="screen shot 2018-10-28 at 9 49 57 am" src="https://user-images.githubusercontent.com/37519914/47966499-6269ce80-e010-11e8-99cd-88c73b6988de.png">

Thanks so much in advance for your help. I _love_ developing with kivy, so I'd be really pleased to be able to release something to the general public with it!

**Comment by Dirk-Sandberg** [comment]
bump

**Comment by tito** [comment]
Are the file correctly 0 bytes? can you check?
Maybe it's something new, i've an app to submit in the next few weeks, i'll be able to solve it then. But you can always search first.

**Comment by Dirk-Sandberg** [comment]
The .so files should be 0 bytes? No, they aren't 0 bytes. If you run into this issue and have some feedback that would be very much welcome - thank you!

**Comment by Dirk-Sandberg** [comment]
Could it have to do with the location of my python modules? I have all my modules copied into the directory of my main python program. Is that the correct place for them to be? I noticed some warning in Xcode about not being able to find the kivy-ios/dist/frameworks folder (it wasn't created automatically by kivy-ios). Maybe that's a clue as well? Have a look:
<img width="772" alt="screen shot 2018-11-22 at 9 59 47 pm" src="https://user-images.githubusercontent.com/37519914/48929038-1f2dad80-eea2-11e8-956c-63acfcf6e050.png">
<img width="270" alt="screen shot 2018-11-22 at 10 03 24 pm" src="https://user-images.githubusercontent.com/37519914/48929093-8c414300-eea2-11e8-9510-48ab135ef9c2.png">

I appreciate your help!

**Comment by Dirk-Sandberg** [comment]
Is it because I haven't built the recipes for some of these modules? In the list of recipes available, not all the modules I need are listed (aka requests, concurrent, pygame). Does this mean I literally cannot use those modules? That would be a major bummer, as I'm using kivy.garden.mapview which depends on concurrent and requests. Please let me know if that's the case!

**Comment by Dirk-Sandberg** [comment]
@tito I finally solved my problem. There were a couple things wrong.

**ISSUE**
My app uses Mapview, and when I tried to run my app the first time it couldn't find the module. To fix this, I had copied my entire 'kivy' module directory (from foo/site-packages/bar on my computer) into my project. That made it so the app could find the Mapview module, but the kivy module has all these '.so' files which caused an issue only when trying to upload the app to the app store.

**SOLUTION**
To fix, I had to use 'garden install --app mapview' while in my project's directory. Then I removed the kivy folder from my projects directory completed. The other modules I make use of (requests, concurrent) needed to stay in my project's directory. I also included the fix from https://github.com/kivy/kivy/issues/4030. SO HAPPY! Can't wait to publish the app!
